airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/query.py

```diff
@@ -1984,3 +1984,174 @@ def record_process_components(self, record: MutableMapping[str, Any]) -> Iterabl
+        Field(name="appStoreAppUrl", alias="app_store_app_url"),
+        Field(name="installUrl", alias="install_url"),
+        Field(name="appStoreDeveloperUrl", alias="app_store_developer_url"),
+        Field(name="isPostPurchaseAppInUse", alias="is_post_purchase_app_in_use"),
+        Field(name="previouslyInstalled", alias="previously_installed"),
+        Field(name="pricingDetailsSummary", alias="pricing_details_summary"),
+        Field(name="pricingDetails", alias="pricing_details"),
+        Field(name="privacyPolicyUrl", alias="privacy_policy_url"),
+        Field(name="publicCategory", alias="public_category"),
+        Field(name="uninstallMessage", alias="uninstall_message"),
+        Field(name="webhookApiVersion", alias="webhook_api_version"),
+        Field(name="shopifyDeveloped", alias="shopify_developed"),
+        Field(name="id", alias="provider_id"),
+        Field(name="failedRequirements", alias="failed_requirements", fields=failed_reqirements_fields),
+        Field(name="feedback", fields=feedback_fields),
+    ]
+
+    assessments_fields: List[Field] = [
+        Field(name="riskLevel", alias="risk_level"),
+        Field(name="facts", fields=["description", "sentiment"]),
+        Field(name="provider", fields=provider_fields),
+    ]
+
+    risk_fields: List[Field] = [
+        "recommendation",
+        Field(name="assessments", fields=assessments_fields),
+    ]
+
+    # main query
+    query_nodes: List[Field] = [
+        "__typename",
+        "updatedAt",
+        Field(name="id", alias="order_id"),
+        Field(name="risk", fields=risk_fields),
+    ]
+
+    record_composition = {
+        "new_record": "Order",
+        # there are no record components provided for this stream.
+    }
+
+    def _process_assessments(self, assessments: Iterable[MutableMapping[str, Any]]) -> Iterable[MutableMapping[str, Any]]:
+        for assessment in assessments:
+            provider = assessment.get("provider", {})
+            if provider:
+                # save and resolve provider id
+                provider["admin_graphql_api_id"] = provider.get("provider_id")
+                provider["provider_id"] = self.tools.resolve_str_id(provider.get("provider_id"))
+        return assessments
+
+    def _has_risk_recommendation(self, recommendation: Optional[str]) -> bool:
+        # if there are no risk recommendation, the value is literally "NONE",
+        # we should skip such record, because there is no risk info for it.
+        no_risk_pattern = "NONE"
+        return recommendation != no_risk_pattern if recommendation else False
+
+    def record_process_components(self, record: MutableMapping[str, Any]) -> Optional[Iterable[MutableMapping[str, Any]]]:
+        """
+        Defines how to process collected components.
+        """
+        # unnest mandatory fields from their placeholders
+        risk = record.get("risk", {})
+        recommendation = risk.get("recommendation") if risk else None
```

Comment:
The validation from L2142 wouldn't work on that way.

```suggestion
        risk = record.get("risk", None)
        recommendation = risk.get("recommendation") if risk else None
```

Reply:
otherwise: 
```suggestion
        risk = record.get("risk", {})
        recommendation = risk.get("recommendation", None)
```

Reply:
It will, because the `{}, [], ()` are treated as `None`, in `python >=3.7`. 

No need to have the statements like this:
```
risk = record.get("risk", None)
```
Because `.get()` function returns `None` by default, if the object doesn't contain the target entity to be get.

Reply:
btw revisiting this, something true is that you can change 
````

recommendation = risk.get("recommendation") if risk else None
```
to:
```
recommendation = risk.get("recommendation")
```
As you have already put a safe `risk = record.get("risk", {})` before if "recommendation" is not present it will return None by default, as you have already stated.

Reply:
The Shopify data has its pitfalls. We might have the literal `None` for the result of this:
```python
risk = record.get("risk", {})
```
If the `risk`, property exists and contains one of: `[None, "None"]`, instead of `objectType ( {} )`

Example:
```python
risk: str | NoneType = "None" | None
risk.get("recommendation")

# Traceback (most recent call last):
#   File "<stdin>", line 1, in <module>
#   AttributeError: 'str' object has no attribute 'get'
```

To avoid such occasions, this check takes place, as well as all other similar checks in other classes for `query.py`:
``` python
recommendation = risk.get("recommendation") if risk else None
```

Problem:
- there is no guarantee and we cannot rely on data enforcement from the server side, we should be able to enforce the data types where possible on our side (source side), this gives us more control over the data flow, in general, and we avoid small bugs we don't want for the `critical source`.

Reply:
So, my last question is, can the **risk variable value** be the string "None"? That's the only thing that confuses me about your last statement:

```
   risk: str  = "None"
   recommendation = risk.get("recommendation") if risk else None
# Traceback (most recent call last):
# File "<stdin>", line 1, in <module>
# AttributeError: 'str' object has no attribute 'get'
```
I mean, str "None" will be evaluated as truthy.

Reply:
Yes, it can). That is the case; we can have it from the server; unfortunately, for some data cases, when the field wasn't populated or not used for Customer. I guess this is the `API changes` legacy we may have at a certain point.